modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -0,0 +1,210 @@
+import org.wso2.charon3.core.exceptions.CharonException;
+import org.wso2.charon3.core.exceptions.InternalErrorException;
+import org.wso2.charon3.core.exceptions.NotFoundException;
+import org.wso2.charon3.core.extensions.UserManager;
+import org.wso2.charon3.core.objects.User;
+import org.wso2.charon3.core.protocol.SCIMResponse;
+import org.wso2.charon3.core.schema.SCIMConstants;
+import org.wso2.charon3.core.schema.SCIMResourceSchemaManager;
+import org.wso2.charon3.core.schema.SCIMResourceTypeSchema;
+import org.wso2.charon3.core.utils.CopyUtil;
+import org.wso2.charon3.core.utils.ResourceManagerUtil;
+
+import java.util.Map;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+/**
+ * Test class of UserResourceManager.
+ */
+@PrepareForTest({AbstractResourceManager.class})
+public class UserResourceManagerTest extends PowerMockTestCase {
+
+    UserResourceManager userResourceManager = new UserResourceManager();
+
+    private User getNewUser() throws BadRequestException, CharonException, InternalErrorException {
+
+        String scimObjectString = "{\n" +
+                "  \"schemas\": \n" +
+                "    [\"urn:ietf:params:scim:schemas:core:2.0:User\",\n" +
+                "    \"urn:ietf:params:scim:schemas:extension:enterprise:2.0:User\"\n" +
+                "    ],\n" +
+                " \"name\": {\n" +
+                "    \"givenName\": \"Rash\",\n" +
+                "    \"familyName\": \"Na\",\n" +
+                "    \"formatted\": \"Rash Na\"\n" +
+                "  },\n" +
+                "  \"userName\": \"rash\",\n" +
+                "  \"password\": \"rash123\",\n" +
+                "  \"id\": \"123\",\n" +
+                "  \"emails\": [\n" +
+                "      {\n" +
+                "        \"type\": \"home\",\n" +
+                "        \"value\": \"rash@gmail.com\",\n" +
+                "         \"primary\": true\n" +
+                "      }\n" +
+                "  ]\n" +
+                "}";
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        User user = (User) decoder.decodeResource(scimObjectString, schema, new User());
+
+        return user;
+    }
+
+    @DataProvider(name = "dataForGetSuccess")
+    public Object[][] dataToGetSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        UserManager userManager = mock(UserManager.class);
+
+        return new Object[][]{
+
+                {id, userManager, "userName", null, 200, user}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetSuccess")
+    public void testGetSuccess(String id, Object objectUserManager, String attributes,
```

Comment:
If we can have a mocked UserManager object for the class, we can get rid of the `Object objectUserManager` parameter in method signatures.

Reply:
fixed in f6c70ca0d7276c5e56af08faad3962e0bb9df4ad